src/main.rs

```diff
@@ -1,29 +1,118 @@
-    let shutdown_signal = async move {
-        // Wait for the CTRL+C signal
-        tokio::signal::ctrl_c()
-            .await
-            .expect("failed to install CTRL+C signal handler");
+fn main() {
+	if let Err(e) = real_main() {
+        io::stderr().write_all(format!("mwixnet server exited with error:\n{}\n", e).as_bytes()).unwrap();
+        std::process::exit(1);
+    }
+
+	std::process::exit(0);
+}
+
+fn real_main() -> std::result::Result<(), Box<dyn std::error::Error>> {
+	let yml = load_yaml!("../mwixnet.yml");
+	let args = App::from_yaml(yml).get_matches();
+    
+    let config_path = match args.value_of("config_file") {
+        Some(path) => PathBuf::from(path),
+        None => {
+            let mut current_dir = env::current_dir()?;
+            current_dir.push("mwixnet-config.toml");
+            current_dir
+        }
     };
+    let password = args.value_of("pass").ok_or(Error::new(ErrorKind::LoadConfigError))?;
+    let password = ZeroingString::from(password);
+
+    let round_time = args.value_of("round_time").map(|t| t.parse::<u32>().unwrap() );
+    let bind_addr = args.value_of("bind_addr");
+    let grin_node_url = args.value_of("grin_node_url");
+    let wallet_owner_url = args.value_of("wallet_owner_url");
+    
+    // Write a new config file if init-config command is supplied
+    if let ("init-config", Some(_)) = args.subcommand() {
+        if config_path.exists() {
+            panic!("Config file already exists at {}", config_path.to_string_lossy());
+        }
+
+        let server_config = ServerConfig {
+            key: secp::random_secret(),
+            interval_s: round_time.unwrap_or(DEFAULT_INTERVAL),
+            addr: bind_addr.unwrap_or("0.0.0.0:3000").parse()?,
+            grin_node_url: grin_node_url.unwrap_or("127.0.0.1:3413").parse()?,
+            wallet_owner_url: wallet_owner_url.unwrap_or("127.0.0.1:3420").parse()?,
+        };
+
+        config::write_config(&config_path, &server_config, &password)?;
+        println!("Config file written to {:?}. Please back this file up in a safe place.", config_path);
+        return Ok(());
+    }
+
+    let mut server_config = config::load_config(&config_path, &password)?;
+
+    // Override bind_addr, if supplied
+    if let Some(bind_addr) = bind_addr {
+        server_config.addr = bind_addr.parse()?;
+    }
+
+    // Override grin_node_url, if supplied
+    if let Some(grin_node_url) = grin_node_url {
+        server_config.grin_node_url = grin_node_url.parse()?;
+    }
+
+    // Override wallet_owner_url, if supplied
+    if let Some(wallet_owner_url) = wallet_owner_url {
+        server_config.wallet_owner_url = wallet_owner_url.parse()?;
+    }
+
+    // Open wallet
+    let wallet_pass = args.value_of("wallet_pass").ok_or(error::Error::new(error::ErrorKind::LoadConfigError))?;
+    let wallet_pass = grin_util::ZeroingString::from(wallet_pass);
+    let wallet = HttpWallet::open_wallet(&server_config.wallet_owner_url, &wallet_pass)?;
```

Comment:
This seems to be calling `open_wallet` by only using the wallet password and without the `api_secret`. I'm not 100% sure, but I was under the impression that to do _any_ kind of requests, you'd need to first call `init_secure_api` and then do encrypted requests as these tests suggest https://github.com/mimblewimble/grin-wallet/blob/master/tests/owner_v3_init_secure.rs#L90-L93
Is it possible to do this without the need to do the "secure dance"?